include/filewriter/sharedqueue.hpp

```diff
@@ -0,0 +1,54 @@
+/**
+ * @file sharedqueue.hpp
+ * @author Jeffin Sam
+ */
+#ifndef MLBOARD_SHARED_QUEUE_HPP
+#define MLBOARD_SHARED_QUEUE_HPP
+
+#include <mlboard/core.hpp>
+
+namespace mlboard {
+
+/**
+ * Class responsible to share a queue among main thread and event logger
+ * thread.
+ * 
+ * @tparam Datatype datatype of the elements queue would be holding. 
+ */
+template <typename Datatype>
+class SharedQueue
+{
+ public:
+  /**
+   * Function to pop an element from the queue.
+   */ 
+  Datatype Pop();
+
+  /**
+   * Function to push an element in the queue.
+   * 
+   * @param item The element to be pushed.
+   */
+  void Push(const Datatype& item);
+  //! Get the size of the queue.
+  size_t Size() const { return queue_.size(); }
+  //! Get the maximum size of the queue.
+  size_t MaxSize() const { return maxSize; }
+  //! Modify the maximum size of the queue.
+  size_t& MaxSize() { return maxSize; }
+  // ! Check if queue is empty.
+  bool Empty() { return queue_.empty(); }
+ private:
+  std::queue<Datatype> queue_;
```

Comment:
Same over here I think comments would lead to a better understanding to a novice.

Reply:
Done